File: python/tests/test_example.py
"""Example test module to verify testing infrastructure works correctly."""

import pytest

# Constants for test values
EXPECTED_SUM: int = 4
EXPECTED_PRODUCT: int = 12
EXPECTED_QUOTIENT: int = 5


def test_basic_arithmetic() -> None:
    """Test basic arithmetic to verify testing framework is working."""
    # Test basic arithmetic operations
    assert EXPECTED_SUM == 2 + 2, "Basic arithmetic should work"
    assert EXPECTED_PRODUCT == 3 * 4, "Multiplication should work"
    assert EXPECTED_QUOTIENT == 10 / 2, "Division should work"

File: matlab/tests/test_example.m
function tests = test_example
%TEST_EXAMPLE Example test suite for MATLAB unit testing
%
%   TESTS = TEST_EXAMPLE() creates a test suite using functiontests framework.
%
%   This is a simple example test suite demonstrating basic unit testing
%   functionality in MATLAB.
%
%   Example:
%       tests = test_example;
%       run(tests);
%
%   See also: functiontests, localfunctions

arguments
end

tests = functiontests(localfunctions);
end

function test_truth(testCase)
%TEST_TRUTH Test basic arithmetic truth
%
%   TEST_TRUTH(TESTCASE) verifies that basic arithmetic operations work correctly.
%
%   Input Arguments:
%   ----------------
%   TESTCASE - Test case object from functiontests framework

arguments
    testCase (1,1) matlab.unittest.TestCase
end

verifyEqual(testCase, 1+1, 2);
end
